_posts/2015-11-13-svg-sprites-angular.md

```diff
@@ -0,0 +1,167 @@
+        <path d="m0 256c0 141 115 256 256 256s256-115 256-256-115-256-256-256-256 115-256 256zm317-79h-39c-4 0-9 6-9 14v28h48v39h-48v119h-46v-119h-41v-39h41v-24c0-33 23-60 55-60h39z"/>
+    </symbol>
+    <symbol id="icon-social-google" viewBox="0 0 512 512">
+        <path d="m232 184c-7-23-17-30-33-30-2 0-4 0-5 1-7 2-13 8-16 16-3 9-3 18-1 29 5 19 19 32 32 32 2 0 3 0 5-1 15-4 24-26 18-47zm24-184c-141 0-256 115-256 256s115 256 256 256 256-115 256-256-115-256-256-256zm-15 366c-16 8-33 9-40 9h-2-1c-11 0-63-3-63-50s57-50 74-50h1c-10-14-8-27-8-27h-4c-6 0-19-1-30-8-13-9-20-23-20-43 0-57 62-60 63-60h62v2c0 7-13 8-21 9-3 1-9 1-11 2 16 8 19 22 19 41 0 23-9 34-18 43-6 5-11 9-11 14 0 6 7 11 14 17 12 11 28 25 28 49s-11 42-32 52zm136-110h-41v40h-25v-40h-41v-26h41v-40h25v40h41zm-168 31h-5c-11 1-22 5-30 12s-12 16-12 25c2 19 22 31 47 29 24-2 40-16 39-35-1-18-17-31-39-31z"/>
+    </symbol>
+    <symbol id="icon-social-twitter" viewBox="0 0 512 512">
+        <path d="m256 0c-141 0-256 115-256 256s115 256 256 256 256-115 256-256-115-256-256-256zm104 210v6c0 67-50 144-143 144-29 0-55-9-77-23 3 1 7 1 12 1 23 0 45-8 62-22-22 0-41-15-47-35 3 1 6 1 10 1s9-1 13-2c-23-4-41-25-41-49v-1c7 4 15 6 23 7-13-10-22-25-22-42 0-10 2-18 7-26 24 31 62 51 104 53-1-4-2-8-2-12 0-27 23-50 51-50 14 0 27 6 36 16 12-2 23-7 33-12-4 11-12 21-23 28 11-2 20-4 29-8-6 10-15 19-25 26z"/>
+    </symbol>
+</svg>
+
+```
+
+And that is already our svg sprite. Now we just have to know how to reference it. Cool thing about svg elements is, that we can reference them by id (for with css sprites, you need to specify 'coordinates' of specific icon as background position), so to get a facebook icon, we can just do this
+
+
+```
+<svg class="svg-icon">
+    <use xlink:href="#icon-social-facebook"></use>
+</svg>
+
+```
+
+We can apply some css properties to it (to svg-icon class) and style it the way we want and that would be it. With that little change, you can improve the way you handle your assets and make it much simpler to use in different context.
+
+
+## Gotchas - is not always roses
+
+- Internet explorer (of course) will not load svgs if you have them in external file (not embedded in template itself)
+- Safari will not load your icons if they are referenced before they are defined (embedded after called)
+- < IE9 is not supporting svgs
+- Using `<base>` changes behavior in firefox (more below)
+
+Currently, we are embedding svg sprites in our template, just after opening `<body>` tag for good cross browser support. If you want to keep you resources out of your template, you can use a great shim written by Jonathan Neal [svg4everybody] (https://github.com/jonathantneal/svg4everybody).
+
+And if you need to support older browsers (IE8 etc.) you can use image as fallback.
+
+## Examples
+
+[Here] (https://github.com/SinisaG/svg-blog-demo) are a few examples that shows some of the browser glitches.
+
+You can run the examples using [5minutefork] (http://5minfork.com/). More in readme....
```

Comment:
:+1: You can also link directly to the fork with http://5minfork.com/SinisaG/svg-blog-demo